### PR TITLE
Update docstrings for 'classes' pkg, submodules

### PR DESCRIPTION
**Pull request by hashknot**

### CODE REVIEW

File: tethne/classes/feature.py [hunk 1]
```diff
@@ -627,20 +706,42 @@ def transform(self, func):
...
         features = {}
         for i, feature in self.features.iteritems():
             features_ = []
             for f, v in feature:
                 t = self.lookup[f]
                 f_ = func(f, v, self.counts[t], self.documentCounts[t])
                 if f_:
-                    feature_.append((f_, v))
-            features[i] = Feature(feature_)
+                    features_.append((f_, v))
```
Comment (hashknot) [inline tethne/classes/feature.py:764]: This change doesn't belong in this commit.

File: tethne/classes/feature.py [hunk 2]
```diff
@@ -627,20 +706,42 @@ def transform(self, func):
...
+            features[i] = Feature(features_)
```
Comment (hashknot) [inline tethne/classes/feature.py:765]: Same here.

**Review by hashknot — COMMENTED** [review]
There are two changes which are not related to this task. I will upload another commit with those changes removed.

File: tethne/classes/corpus.py [hunk 3]
```diff
@@ -561,10 +611,28 @@ def slice(self, window_size=1, step_size=1, cumulative=False,
...
+            (default: ``True``) If ``False``\, second value in each generator
+            yielded tuple is a list of :class:`Paper`\s.
+        feature_name : str
+            (default: ``None``) If not ``None``\, second value in each generated
+            yielded tuple is a :class::``FeatureSet`` instance.
 
         Returns
         -------
-        generator
+        A generator that yields ``(key, subcorpus)`` tuples for
```
Comment (erickpeirson) [inline tethne/classes/corpus.py]: Try to use a format similar to ``Parameters`` here, e.g.

```
Returns
---------
generator
    A generator that yields ``(key, subcorpus)`` tuples....
```

File: tethne/classes/corpus.py [hunk 4]
```diff
@@ -713,6 +785,20 @@ def top_features(self, featureset_name, topn=20, by='counts',
...
+        Parameters
+        ----------
+        selector : object
+            Refer :meth:`.Corpus.select` for argument description.
+        featureset_name : str
+            Name of a :class:`.FeatureSet` in the :class:`.Corpus`\.
+
+        Returns
+        -------
+        An instance of :class:`FeatureSet`.
```
Comment (erickpeirson) [inline tethne/classes/corpus.py]: See above

File: tethne/classes/corpus.py [hunk 5]
```diff
@@ -733,6 +819,14 @@ def subcorpus(self, selector):
...
            <tethne.classes.corpus.Corpus object at 0x10278ea10>
 
+        Parameters
+        ----------
+        selector : object
+            Refer :meth:`.Corpus.select` for argument description.
+
+        Returns
+        -------
+        :class:`Corpus` object.
```
Comment (erickpeirson) [inline tethne/classes/corpus.py]: See above

File: tethne/classes/feature.py [hunk 6]
```diff
@@ -71,6 +71,10 @@ def __init__(self, tokens, contexts=None, reference=None):
     def unique(self):
         """
         The `set` of unique elements in this :class:`.Feature`\.
+
+        Returns
+        -------
+        elements : set
```
Comment (erickpeirson) [inline tethne/classes/feature.py]: See above; just 

```
Returns
-------
set
```

File: tethne/classes/feature.py [hunk 7]
```diff
@@ -290,13 +295,24 @@ def extend(self, data):
...
         return set()
 
     @property
     def norm(self):
+        """
+        Return normalized feature values so that all values sum to 1.
+
+        Returns
+        -------
+        :class:`Feature` object with normalized values.
```
Comment (erickpeirson) [inline tethne/classes/feature.py]: See above

File: tethne/classes/feature.py [hunk 8]
```diff
@@ -324,6 +340,9 @@ def top(self, topn=10):
         return [self[i] for i in argsort(list(zip(*self))[1])[::-1][:topn]]
 
     def value(self, element):
+        """
+        Get the value for ``element`` from this :class:`Feature` object.
```
Comment (erickpeirson) [inline tethne/classes/feature.py:344]: Parameters

File: tethne/classes/feature.py [hunk 9]
```diff
@@ -368,13 +394,27 @@ def unique(self):
...
+        """
+        Count of documents in this :class:`BaseFeatureSet` object.
+        """
         return len(self.features)
 
     def count(self, elem):
+        """
+        Get count of ``elem`` across all documents in the
+        :class:`BaseFeatureSet` object.
+
```
Comment (erickpeirson) [inline tethne/classes/feature.py:432]: Parameters

File: tethne/classes/feature.py [hunk 10]
```diff
@@ -385,15 +425,37 @@ def count(self, elem):
             return 0.
 
     def documentCount(self, elem):
+        """
+        Get count of documents containing ``elem`` element.
+
```
Comment (erickpeirson) [inline tethne/classes/feature.py:453]: Parameters

File: tethne/classes/feature.py [hunk 11]
```diff
@@ -385,15 +425,37 @@ def count(self, elem):
...
+        """
         if elem in self.lookup:
             return self.documentCounts[self.lookup[elem]]
         else:
             return 0.
 
     def papers_containing(self, elem):
+        """
+        Get list of papers containing ``elem`` element.
+
```
Comment (erickpeirson) [inline tethne/classes/feature.py:470]: Parameters

File: tethne/classes/feature.py [hunk 12]
```diff
@@ -627,20 +706,42 @@ def transform(self, func):
...
+            features[i] = Feature(features_)
         return FeatureSet(features)
 
     def as_matrix(self):
         """
+        Return the features and documents as a matrix.
 
+        Returns
+        -------
+        Returns  NxM matrix in the form of list of lists, where N is the number
```
Comment (erickpeirson) [inline tethne/classes/feature.py]: See above

File: tethne/classes/streaming.py [hunk 13]
```diff
@@ -7,13 +7,19 @@
 # TODO: persist the index and data.
 
 class StreamingIndex(object):
+    """
+    'dict' like class supporting memory-friendly index.
+    """
     def __init__(self, name='index', base_path='.', serializer=pickle):
         """
-
         Parameters
```
Comment (erickpeirson) [inline tethne/classes/streaming.py]: Put the parameters in the class docstring

File: tethne/classes/streaming.py [hunk 14]
```diff
@@ -102,6 +111,9 @@ def __iter__(self):
         return PList(self)
 
     def __init__(self, *args, **kwargs):
+        """
+        Refer :meth:`.Corpus.__init__` for argument description and usage.
```
Comment (erickpeirson) [inline tethne/classes/streaming.py]: Sphinx ignores the constructor when auto-generating API documentation, so this kind of thing should go in the class docstring.

**Review by erickpeirson — COMMENTED** [review]
@hashknot This is excellent! Just a few small things, but otherwise fantastic.

It's not the end of the world if there are small changes to the actual code -- just be sure to reference the relevant bug/story in the pull request.

**Comment by hashknot** [comment]
Fixed the "Return" docstrings. Added missing parameter docstrings.
